src/amplitude-client.js

```diff
@@ -1897,4 +1978,157 @@ AmplitudeClient.prototype._refreshDynamicConfig = function _refreshDynamicConfig
+  if (!utils.validateInput(eventUploadThreshold, 'eventUploadThreshold', 'number')) {
+    return;
+  }
+
+  if (this._shouldDeferCall()) {
+    return this._q.push(['setEventUploadThreshold'].concat(Array.prototype.slice.call(arguments, 0)));
+  }
+
+  try {
+    this.options.eventUploadThreshold = eventUploadThreshold;
+  } catch (e) {
+    utils.log.error(e);
+  }
+};
+
+/**
+ * Dynamically adjust server URL
+ * @public
+ * @param {bool} useDynamicConfig - if enable dynamic config or not.
+ * @example amplitudeClient.setUseDynamicConfig(true);
+ */
+AmplitudeClient.prototype.setUseDynamicConfig = function setUseDynamicConfig(useDynamicConfig) {
+  if (!utils.validateInput(useDynamicConfig, 'useDynamicConfig', 'boolean')) {
+    return;
+  }
+
+  if (this._shouldDeferCall()) {
+    return this._q.push(['setUseDynamicConfig'].concat(Array.prototype.slice.call(arguments, 0)));
+  }
+
+  try {
+    this.options.useDynamicConfig = useDynamicConfig;
+    this._refreshDynamicConfig();
+  } catch (e) {
+    utils.log.error(e);
+  }
+};
+
+/**
+ * Sets the server zone , used for server api endpoint and dynamic configuration..
+ * @public
+ * @param {string} serverZone - the server zone value. AmplitudeServerZone.US or AmplitudeServerZone.EU.
+ * @param {bool} serverZoneBasedApi - (optional) update api endpoint with serverZone change or not. For data residency, recommend to enable it unless using own proxy server.
+ * @example amplitudeClient.setServerZone('joe@gmail.com', true);
+ */
+AmplitudeClient.prototype.setServerZone = function setServerZone(serverZone, serverZoneBasedApi = true) {
+  if (
+    (serverZone !== AmplitudeServerZone.EU && serverZone !== AmplitudeServerZone.US) ||
+    !utils.validateInput(serverZoneBasedApi, 'serverZoneBasedApi', 'boolean')
+  ) {
+    return;
+  }
+
+  if (this._shouldDeferCall()) {
+    return this._q.push(['setServerZone'].concat(Array.prototype.slice.call(arguments, 0)));
+  }
+
+  try {
+    this.options.serverZone = serverZone;
+    this.options.serverZoneBasedApi = serverZoneBasedApi;
+    if (serverZoneBasedApi) {
+      this.options.apiEndpoint = getEventLogApi(this.options.serverZone);
+    }
+  } catch (e) {
+    utils.log.error(e);
+  }
+};
+
+/**
+ * Sets the server URL for the request.
+ * @public
+ * @param {string} serverUrl - The value of the server URL.
+ * @example amplitudeClient.setServerUrl('api.amplitude.com');
+ */
+AmplitudeClient.prototype.setServerUrl = function setServerUrl(serverUrl) {
+  if (!utils.validateInput(serverUrl, 'serverUrl', 'string')) {
+    return;
+  }
+
+  if (this._shouldDeferCall()) {
+    return this._q.push(['setServerUrl'].concat(Array.prototype.slice.call(arguments, 0)));
+  }
+
+  try {
+    this.options.apiEndpoint = serverUrl;
+  } catch (e) {
+    utils.log.error(e);
+  }
+};
+
+/**
+ * Upload all unsent events.
+ * @public
+ * @example amplitudeClient.uploadEvents();
+ */
+AmplitudeClient.prototype.uploadEvents = function uploadEvents() {
```

Comment:
Do we really need this method? Can flutter just call `sendEvents`?

Reply:
Aha, you are right. It's redundant to create another wrapper.

Reply:
Do you want to remove this method?